artifactory/commands/golang/gopublish.go

```diff
@@ -53,6 +54,20 @@ func (gpc *GoPublishCommand) SetPublishPackage(publishPackage bool) *GoPublishCo
 	return gpc
 }
 
+func (gpc *GoPublishCommand) SetConfigFilePath(configFilePath string) (err error) {
```

Comment:
This function is different from the ```SetConfigFilePath``` on other commands. For example:
```
func (npc *NpmPublishCommand) SetConfigFilePath(configFilePath string) *NpmPublishCommand {
	npc.configFilePath = configFilePath
	return npc
}
```
If I understand the code correctly, this function aims to serve both the old and new forms of the go-publish command.
I suggest however to split this into two different commands - GoLegacyPublishCommand and GoPublishCommand. We took this approach for other commands, which we needed to modify. This should make it very easy to completely remove the legacy commands in the future, as part of jfrog-cli 2.0.

Similarly, thre deprecated ```publishPackage```, can alsoi move to the legacy commabnd.